src/main/java/com/google/devtools/build/lib/analysis/starlark/FunctionTransitionUtil.java

```diff
@@ -127,21 +128,22 @@ static ImmutableMap<String, BuildOptions> applyAndValidate(
    * <p>Transitions can also explicitly set --platforms to be clear what platform they set.
    *
    * <p>Platform mappings: https://bazel.build/concepts/platforms-intro#platform-mappings.
-   *
-   * <p>This doesn't check that the changed value is actually different than the source (i.e.
-   * setting {@code --cpu=foo} when {@code --cpu} is already {@code foo}). That could unnecessarily
-   * fork configurations that are really the same. That's a possible optimization TODO.
    */
   private static Map<String, Object> handleImplicitPlatformChange(
-      Map<String, Object> originalOutput) {
-    boolean changesCpu = originalOutput.containsKey(COMMAND_LINE_OPTION_PREFIX + "cpu");
-    boolean changesPlatforms = originalOutput.containsKey(COMMAND_LINE_OPTION_PREFIX + "platforms");
-    return changesCpu && !changesPlatforms
-        ? ImmutableMap.<String, Object>builder()
-            .putAll(originalOutput)
-            .put(COMMAND_LINE_OPTION_PREFIX + "platforms", ImmutableList.<Label>of())
-            .build()
-        : originalOutput;
+      BuildOptions options, Map<String, Object> originalOutput) {
+    Object newCpu = originalOutput.get(COMMAND_LINE_OPTION_PREFIX + "cpu");
```

Comment:
Nit: The names `newCpu` and `originalOutput` don't seem to match. Can we rename one of them?

Reply:
Done, renamed the existing variable to be more descriptive.